src/lapack/dec/q/qrut/front/flamec/FLASH_QR_UT_form_Q.c

```diff
@@ -0,0 +1,81 @@
+/*
+
+    Copyright (C) 2014, The University of Texas at Austin
+
+    This file is part of libflame and is available under the 3-Clause
+    BSD license, which can be found in the LICENSE file at the top-level
+    directory, or at http://opensource.org/licenses/BSD-3-Clause
+
+*/
+
+#include "FLAME.h"
+
+FLA_Error FLASH_QR_UT_form_Q( FLA_Obj A, FLA_Obj T, FLA_Obj Q )
+{
+    FLA_Error r_val = FLA_SUCCESS;
+    FLA_Obj   QTL, QTR,
+              QBL, QBR;
+    FLA_Obj   W;
+    dim_t     b;
+
+    if ( FLA_Check_error_level() >= FLA_MIN_ERROR_CHECKING )
+        FLA_QR_UT_form_Q_check( A, T, Q );
+
+    if ( FLA_Obj_is_overlapped( A, Q ) == FALSE )
+    {
+        // If A and Q are different objects, Q is explicitly formed with A.
+
+        // Set Q identify
+        FLA_Set_to_identity( Q );
+
+        // Q = H_{0} H_{1} ... H_{k-1}
+        FLASH_Apply_Q_UT_create_workspace_side( FLA_LEFT, T, Q, &W );
+        r_val = FLA_Apply_Q_UT( FLA_LEFT, FLA_NO_TRANSPOSE,
+                                FLA_FORWARD, FLA_COLUMNWISE,
+                                A, T, W, Q );
+        //FLASH_Obj_free( &W );  Need to see if it's alright to sub these out, this might cause a mem leak
```

Comment:
Had to switch this to _without_buffer. Not sure if this is correct, but I was getting a undefined mem reference when trying to free W with this one.